actors/evm/src/interpreter/instructions/log.rs

```diff
@@ -1,28 +1,57 @@
+use crate::interpreter::instructions::memory::get_memory_region;
+use fvm_ipld_encoding::{to_vec, RawBytes};
+use fvm_shared::event::{Entry, Flags};
 use {
     crate::interpreter::{ExecutionState, StatusCode, System},
     fil_actors_runtime::runtime::Runtime,
 };
 
-#[cfg(debug_assertions)]
-pub fn log(
-    _state: &mut ExecutionState,
-    _system: &System<impl Runtime>,
-    _num_topics: usize,
-) -> Result<(), StatusCode> {
-    todo!("unimplemented");
-}
+/// The event key for the Ethereum log data.
+const EVENT_DATA_KEY: &str = "data";
+
+/// The event keys for the Ethereum log topics.
+const EVENT_TOPIC_KEYS: &[&str] = &["topic1", "topic2", "topic3", "topic4"];
 
-#[cfg(not(debug_assertions))]
 #[inline]
 pub fn log(
     state: &mut ExecutionState,
-    _system: &System<impl Runtime>,
+    system: &System<impl Runtime>,
     num_topics: usize,
 ) -> Result<(), StatusCode> {
-    // TODO: Right now, we just drop everything. But we implement this in production anyways so
-    // things work.
-    for _ in 0..num_topics {
-        state.stack.pop();
+    // Handle the data.
+    // Passing in a zero-sized memory region omits the data key entirely.
+    // LOG0 + a zero-sized memory region emits an event with no entries whatsoever. In this case,
+    // the FVM will record a hollow event carrying only the emitter actor ID.
```

Comment:
why is this useful?

Reply:
Ethereum can record events with no topics nor data; the only useful thing is the emitter address in that case. Not sure how much it gets used in practice, but it is possible with that runtime. So not doing it here would break compatibility.